exchange.go

```diff
@@ -15,6 +15,7 @@ import (
 
 const (
 	maxRequeueNo            = 3
+	defaultMaxConnections   = 20
```

Comment:
This is misleading as the number is the max number of idle connections in the pool. The total number of connections can be larger. We might want a way to limit that, maybe having another chan for connection currently in use.